testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -347,35 +347,44 @@ fn query_sequence_numbers(
     client: &AdmissionControlClient,
     addresses: &[AccountAddress],
 ) -> failure::Result<Vec<u64>> {
-    let mut update_request = UpdateToLatestLedgerRequest::default();
-    for address in addresses {
-        let mut request_item = RequestItem::default();
-        let mut account_state_request = GetAccountStateRequest::default();
-        account_state_request.address = address.to_vec();
-        request_item.requested_items = Some(RequestedItems::GetAccountStateRequest(
-            account_state_request,
-        ));
-        update_request.requested_items.push(request_item);
-    }
-    let resp = client
-        .update_to_latest_ledger(&update_request)
-        .map_err(|e| format_err!("update_to_latest_ledger failed: {:?} ", e))?;
-    let mut result = Vec::with_capacity(resp.response_items.len());
-    for item in resp.response_items.into_iter() {
-        let item = ResponseItem::try_from(item)
-            .map_err(|e| format_err!("ResponseItem::from_proto failed: {:?} ", e))?;
-        if let ResponseItem::GetAccountState {
-            account_state_with_proof,
-        } = item
-        {
-            let account_resource = get_account_resource_or_default(&account_state_with_proof.blob)
+    let mut result = vec![];
+    let mut i = 0;
+    while i < addresses.len() {
```

Comment:
You can also use [slice:chunk](https://doc.rust-lang.org/std/primitive.slice.html#method.chunks) for similar effect
Up to you, just mentioning it exists

@bors-libra delegate+

Reply:
I was trying to find something like this in the `Iterator` trait..